libraries/ESP8266WebServer/src/Parsing.cpp

```diff
@@ -317,8 +313,8 @@ void ESP8266WebServer::_parseArguments(String data) {
       continue;
     }
     RequestArgument& arg = _currentArgs[iarg];
-    arg.key = data.substring(pos, equal_sign_index);
-	arg.value = data.substring(equal_sign_index + 1, next_arg_index);
+    arg.key = urlDecode(data.substring(pos, equal_sign_index));
```

Comment:
Why decode the key as well, any use case ?

Reply:
Any case that uses non-URL-safe characters in key names is a use case for this. They need to be decoded as well as values.